# Lit Review

## Geographic and Genomic Distribution of SARS-CoV-2 Mutations

Author: Daniele Mercatelli and Federico M. Giorgi
Dept. of Pharmacy and Biotechnology, University of Bologna, Bologna, Italy

Summary:

This paper analyzes the 48,635 SARS-CoV-2 complete genomic sequences from GISAID consortium on their mutation in comparison to the original genome from Wuhan, China (NC_045512.2). After aligning the sequences, there are 353,341 mutation events in total, which results in an average of 7.23 mutations per sequence, a mode of 6 mutations, and very few samples have more than 15 events. This low mutation rate means the Covid-19 viruses do not differ significantly from continent to continent; however, from countries to countries, there is telling distinctions. Looking at the mutations themselves, the authors found that most of the mutations (58.2%) are single-nucleotide polymorphisms (SNPs), with half of those are silent SNPs and a quarter are mutations in the intergenic regions. Deletions and insertions are relatively rare, accounting for a total of less than 1%. Among the SNPs, the most prevalent mutation (worldwide) is a transversion affecting the 23,403rd nucleotide: A>G, defining the so-called G-clade of the virus. This strand is prevalent in Europe, Oceanic, South America and Africa. This mutation caused a change in the Spike protein, which is responsible for the initial entry of the virus into the cell. The other two major clades are called "S", after the mutation in ORF8 L84S (Ceraolo and Giorgi, 2020), and "V", from the ORF3a:G251V mutation (co-occuring with NSP6:L37F). Geographically, the G and DR clades are mostly present in Europe, while S and GH are prevalent in the Americas. The "L" clade (the orginal lineage) is represented largely by countries in Asia. Over time, however, the G clade genomes and its derivates, GH and GR, has been become increasingly more common, while clade L, V, and S genomes are gradually disappearing. 


## Genome sequencing of sewage detects regionally prevalent SARS-CoV-2 variants

Author: Alexander Crits-Christophâ€‹ et. al
Multiple Dept. at University of California, Berkeley, CA, USA

Quantitative reverse transcription of polymerase chain reaction (`RT-qPCR`) has been applied to municipal wastewater around the world to detect and quantify the presence and abundance of SARS-CoV-2. If the virus is present, then metatranscriptomic sequencing is used to profile its genetic diversity across infected communities. The researchers in this paper analyzed the RNA from sewage collected by municipal utility districts in the San Francisco Bay Area to generate complete and near-complete SARS-CoV-2 genomes. Through their work, they have demonstrated that the genomes of the "local" virus (in the Bay Area) are more similar to those in California than those outside of the state. Hence, they have provided evidence that their technique can perform "epidemiological surveillance", which can aid in tracking population movement and new genomic mutations of the virus.